### PR TITLE
Config replacers scope clash with config import

### PR DESCRIPTION
**Pull request by alex-dukhno**
This PR contains fix for config replacers and config imports for xml based configuration files (as reported in #15452)

@blazember, could you please take a look? (cc @neilstevenson)

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/AbstractXmlConfigBuilder.java [hunk 1]
```diff
@@ -92,46 +92,69 @@ public Iterator getPrefixes(String namespaceURI) {
     }
 
     protected void process(Node root) throws Exception {
+        replaceImports(root);
         traverseChildrenAndReplaceVariables(root);
-        replaceImportElementsWithActualFileContents(root);
     }
 
-    private void replaceImportElementsWithActualFileContents(Node root) throws Exception {
+    private void replaceVariables(Node root) throws Exception {
```
Comment (blazember) [inline hazelcast/src/main/java/com/hazelcast/config/AbstractXmlConfigBuilder.java]: This and `traverseChildrenAndReplaceVariables(root)` seem to be duplicate methods. Could this be a leftover?

File: hazelcast/src/test/java/com/hazelcast/config/XmlConfigImportVariableReplacementTest.java [hunk 2]
```diff
@@ -137,107 +218,94 @@ public void testImportedConfigVariableReplacement() throws Exception {
...
-        writeStringToStreamAndClose(new FileOutputStream(config1), config1Xml);
-        writeStringToStreamAndClose(new FileOutputStream(config2), config2Xml);
-        writeStringToStreamAndClose(new FileOutputStream(config3), config3Xml);
-        expectInvalid();
-        buildConfig(config1Xml, null);
+        String xmlWithCyclicImport = createFilesWithCycleImports(
+                createFileWithContent("hz1", "xml", ""),
+                createFileWithContent("hz2", "xml", ""),
+                createFileWithContent("hz3", "xml", "")
+        );
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/config/XmlConfigImportVariableReplacementTest.java]: Nice cleanup :+1:

File: hazelcast/src/test/java/com/hazelcast/config/test/builders/MapXmlStoreConfigBuilder.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        this.enabled = true;
+        return this;
+    }
+
+    public MapXmlStoreConfigBuilder disabled() {
+        this.enabled = false;
+        return this;
+    }
+
+    public MapXmlStoreConfigBuilder withInitialMode(String initialMode) {
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/config/test/builders/MapXmlStoreConfigBuilder.java]: You can use the `MapStoreConfig.InitialLoadMode` enum here.

**Comment by alex-dukhno** [comment]
@blazember I added tests to `XmlClientConfigImportVariableReplacementTest`, did some cleanup (I'll make some more in #15824) and rebased over latest master.

Thank you for your review.

**Comment by blazember** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/IOUtils.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package com.hazelcast;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.function.Function;
+
+public class IOUtils {
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/IOUtils.java]: This helper feels to be specific to config, I think it should be under `com.hazelcast.config.helpers`. We have `DeclarativeConfigFileHelper` there for similar reasons. It doesn't use static methods, but I think these methods can go there. Could you please check if we can add them there?

**Comment by blazember** [comment]
@alex-dukhno Thank you for your quick response. I added one more comment for consideration. Also, there are a few checkstyle errors, you can check them with 
```
mvn clean install -DskipTests -Pcheckstyle,spotbugs
```
in the project root. Could you please take a look?

**Comment by blazember** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/config/AbstractXmlConfigBuilder.java (hunk 1 above)
Comment (blazember) [reply hazelcast/src/main/java/com/hazelcast/config/AbstractXmlConfigBuilder.java]: Now `traverseChildrenAndReplaceVariables` is an unused method. Please don't forget to remove it in #15824.

**Comment by blazember** [comment]
Thank you for the fix @alex-dukhno, a really nice PR :+1:

**Pull request merged**